### PR TITLE
Support neosegment displays

### PR DESCRIPTION
**Pull request by rm-hull**
See #106 

Rendered documentation here: http://luma-led-matrix.readthedocs.io/en/feature-neosegment/python-usage.html#neosegments

### CODE REVIEW

File: doc/python-usage.rst [hunk 1]
```diff
@@ -341,6 +341,41 @@ a translation mapping is required, as follows:
 
 This should animate a green dot moving left-to-right down each line.
 
+NeoSegments
+"""""""""""
+`@msurguy <https://twitter.com/msurguy?lang=en>`_ has `crowdsourced some WS2812 neopixels <https://www.crowdsupply.com/maksmakes/neosegment>`_ 
+into a modular 3D-printed seven-segment module: To program these devices:
```
Comment (thijstriemstra) [inline doc/python-usage.rst]: Replace `seven-segment module:` with `seven-segment module.`.

File: doc/python-usage.rst [hunk 2]
```diff
@@ -341,6 +341,41 @@ a translation mapping is required, as follows:
...
+
+    # Set fourth and fifth chars ('S','E') accordingly
+    neoseg.color[3:5] = ["cyan", "blue"]
+    time.sleep(1)
+
+    # Set the entire string to green
+    neoseg.color = "green"
+
+The :py:class:`~luma.led_matrix.device.neosegment` class extends :py:class:`~luma.core.virtual.sevensegment`,
+so the same text assignment (python slicing paradigms) can be use here as well -
```
Comment (thijstriemstra) [inline doc/python-usage.rst]: - Python with a capital
- 'use' should be 'used'

File: examples/neosegment_demo.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+# See LICENSE.rst for details.
+
+import time
+import random
+import colorsys
+from luma.led_matrix.device import neosegment
+from neopixel_demo import gfx
+
+
+def rainbow(n=1000, saturation=1, value=1):
```
Comment (thijstriemstra) [inline examples/neosegment_demo.py]: a docstring for this method would be nice.

File: luma/led_matrix/device.py [hunk 4]
```diff
@@ -393,3 +395,89 @@ def contrast(self, value):
...
+
+
+class neosegment(sevensegment):
+    """
+    Extends the :py:class:`~luma.core.virtual.sevensegment` class specifically
+    for @msurguy's pluggable NeoSegments. It uses the same underlying render
+    techniques as the base class, but provides additional functionality to be
+    able to adddress individual characters colors.
+
+    :param width: the number of 7-segment elements that are cascaded.
```
Comment (thijstriemstra) [inline luma/led_matrix/device.py]: 'the' should be 'The'.

File: tests/test_neosegment.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2014-17 Richard Hull and contributors
```
Comment (thijstriemstra) [inline tests/test_neosegment.py]: 2017 only.

File: tests/test_neosegment.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+# See LICENSE.rst for details.
+
+import pytest
```
Comment (thijstriemstra) [inline tests/test_neosegment.py]: (optional..) missing docstring for this module.

File: doc/python-usage.rst (hunk 1 above)
Comment (rm-hull) [inline doc/python-usage.rst]: done

File: doc/python-usage.rst (hunk 2 above)
Comment (rm-hull) [inline doc/python-usage.rst]: done

File: examples/neosegment_demo.py (hunk 3 above)
Comment (rm-hull) [inline examples/neosegment_demo.py]: done

File: luma/led_matrix/device.py (hunk 4 above)
Comment (rm-hull) [inline luma/led_matrix/device.py]: done

File: tests/test_neosegment.py (hunk 5 above)
Comment (rm-hull) [inline tests/test_neosegment.py]: done

**Comment by rm-hull** [comment]
luma.led_matrix **0.11.0** released with the feature included

**Pull request merged**